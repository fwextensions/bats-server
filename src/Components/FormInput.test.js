import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FormInput from './FormInput';

test('Snapshot', () => {
  const component = renderer.create(
    <FormInput
      label="Unit #"
      onChange={() => {}}
      property="ambulanceIdentifier"
      required
      size="medium"
    />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('FormInput should trigger the onChange callback', () => {
  const mock = jest.fn();

  const wrapper = render(
    <FormInput
      label="Unit"
      onChange={mock}
      property="ambulanceIdentifier"
      required
      size="medium"
    />);

    const input = wrapper.getByLabelText('Unit');
    fireEvent.change(input, { target: { value: "d" }});
    fireEvent.change(input, { target: { value: "a" }});
    fireEvent.change(input, { target: { value: "y" }});
    expect(mock.mock.calls.length).toBe(3)
})