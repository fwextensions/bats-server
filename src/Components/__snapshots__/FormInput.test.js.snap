// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Array [
  <label
    className="usa-label usa-label--required"
    htmlFor="ambulanceIdentifier"
  >
    Unit #
  </label>,
  <div
    className="grid-row flex-align-center"
  >
    <input
      className="usa-input usa-input--medium"
      disabled={false}
      id="ambulanceIdentifier"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>,
]
`;
